xpayments_cloud.py

```diff
@@ -1,565 +1,249 @@
-
-    def do_accept(self, xpid):
+    def do_accept(self, xpid: str) -> str:
         """
         Accept payment
-
-        Parameters
-        ----------
-        xpid: str
-            X-Payments payment ID
-
-        Returns
-        -------
-        list
         """
+        return self.do_action(action='accept', xpid=xpid)
 
-        return self.do_action("accept", xpid)
-
-    def do_decline(self, xpid):
+    def do_decline(self, xpid: str) -> str:
         """
         Decline payment
-
-        Parameters
-        ----------
-        xpid: str
-            X-Payments payment ID
-
-        Returns
-        -------
-        list
         """
+        return self.do_action(action='decline', xpid=xpid)
 
-        return self.do_action("decline", xpid)
-
-    def do_get_info(self, xpid):
+    def do_get_info(self, xpid: str) -> str:
         """
         Get detailed payment information
-
-        Parameters
-        ----------
-        xpid: str
-            X-Payments payment ID
-
-        Returns
-        -------
-        list
         """
+        return self.do_action(action='get_info', xpid=xpid)
 
-        return self.do_action("get_info", xpid)
-
-    def do_get_customer_cards(self, customer_id, status="any"):
+    def do_get_customer_cards(self, customer_id: str, status: str = 'any') -> str:
         """
         Get customer's cards
-
-        Parameters
-        ----------
-        customer_id: str
-            X-Payments customer ID
-        status: str
-            Status of the cards, "any" or "active"
-
-        Returns
-        -------
-        list
         """
-
         params = {
             "customerId": customer_id,
             "status": status
         }
+        return self.request.send(controller='customer', action='get_cards', request_data=params)
 
-        request = Request(self.account, self.api_key, self.secret_key)
-
-        return request.send('customer', 'get_cards', params)
-
-    def do_add_bulk_operation(self, operation, xpids):
+    def do_add_bulk_operation(self, operation: str, xpids: list[str]) -> str:
         """
         Add bulk operation
-
-        Parameters
-        ----------
-        operation: str
-            Bulk operation type
-        xpids: dict
-            List of X-Payments payments ID's
-
-        Returns
-        -------
-        list
         """
-
         params = {
-            'operation': operation,
-            'payments': [],
+            "operation": operation,
+            "payments": [{'xpid': xpid} for xpid in xpids],
         }
+        return self.request.send(controller='bulk_operation', action='add', request_data=params)
 
-        for xpid in xpids:
-            params['payments'].append({'xpid': xpid})
-
-        request = Request(self.account, self.api_key, self.secret_key)
-
-        return request.send('bulk_operation', 'add', params)
-
-    def do_start_bulk_operation(self, batch_id):
+    def do_start_bulk_operation(self, batch_id: str) -> str:
```

Comment:
Returns string?

Reply:
Yes, it looks like the json decoder in send() can return different types, including dict, list and str. I will correct this.